File: proyecto-adoptapet/adoptapet-frontend/src/app/models/solicitud.ts
import { Mascota } from './mascota';

export interface ProgramacionEntrega {
  id_programacion?: number;
  fecha_entrega?: string;
  hora_inicio?: string;
  hora_fin?: string;
  fecha_limite_recojo?: string;
  estado_programacion?: string;
  motivo_cancelacion?: string;
  observacion?: string;
}

export interface Solicitud {
  id_solicitud?: number;
  fecha_registro?: string;
  adoptante?: any;
  mascota?: Mascota;
  comentario?: string;
  acta_pdf?: string;
  motivo_adopcion?: string;
  tipo_vivienda?: string;
  experiencia_mascotas?: string;
  otras_mascotas?: string;
  cantidad_personas_hogar?: number;
  comentarios_adicionales?: string;
  ruta_dni?: string;
  ruta_domicilio?: string;
  ruta_acta_firmada?: string;
  cantidad_reprogramaciones?: number;
  motivo_contingencia?: string;
  fecha_cierre_contingencia?: string;
  estado_solicitud?: string;
  motivo_rechazo?: string;
  programacionEntrega?: ProgramacionEntrega;
}

File: proyecto-adoptapet/adoptapet-backend/src/main/java/com/proyecto/adoptapet/model/Solicitud.java
package com.proyecto.adoptapet.model;

import java.sql.Date;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Solicitud {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id_solicitud;
	
	private Date fecha_registro;
	
	@OneToOne(fetch = FetchType.LAZY)
	@JoinColumn(name="id_adoptante")
	private Adoptante adoptante;

	@OneToOne(fetch = FetchType.LAZY)
	@JoinColumn(name="id_mascota")
	private Mascota mascota;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "id_trabajador")
	private Trabajador trabajador;
	
	private String comentario;
	
	@Column(name="ruta_pdf_acta")
	private String acta_pdf;

	private String motivo_adopcion;

	private String tipo_vivienda;

	private String experiencia_mascotas;

	private String otras_mascotas;

	private Integer cantidad_personas_hogar;

	private String comentarios_adicionales;

	private String ruta_dni;

	private String ruta_domicilio;

	private String ruta_acta_firmada;

	private Integer cantidad_reprogramaciones = 0;

	private String motivo_contingencia;

	private java.time.LocalDateTime fecha_cierre_contingencia;
	
	@Column(name = "estado_solicitud")
	private String estado_solicitud;

	private String motivo_rechazo;

	@OneToOne(mappedBy = "solicitud", fetch = FetchType.LAZY)
	private ProgramacionEntrega programacionEntrega;

}
